src/main/kotlin/org/jitsi/nlj/transform/node/outgoing/ProbingDataSender.kt

```diff
@@ -49,43 +51,40 @@ class ProbingDataSender(
     private val rtxDataSender: PacketHandler,
     private val garbageDataSender: PacketHandler,
     private val diagnosticContext: DiagnosticContext,
-    streamInformationStore: StreamInformationStore
+    private val streamInformationStore: ReadOnlyStreamInformationStore
 ) : EventHandler, NodeStatsProducer {
 
     private val timeSeriesLogger = TimeSeriesLogger.getTimeSeriesLogger(this.javaClass)
     private val logger = getLogger(this.javaClass)
 
-    private var rtxSupported = false
-    private val videoPayloadTypes = mutableSetOf<VideoPayloadType>()
+    // TODO: this will move to stream information store
+    private val videoPayloadTypes: MutableMap<Int, VideoPayloadType> = ConcurrentHashMap()
     private var localVideoSsrc: Long? = null
 
     // Stats
     private var numProbingBytesSentRtx: Int = 0
     private var numProbingBytesSentDummyData: Int = 0
 
     init {
-        streamInformationStore.onRtpPayloadTypesChanged { currentRtpPayloadTypes ->
-            if (currentRtpPayloadTypes.isEmpty()) {
-                videoPayloadTypes.clear()
-                rtxSupported = false
-            } else {
-                currentRtpPayloadTypes.values.forEach { pt ->
-                    if (!rtxSupported && pt is RtxPayloadType) {
-                        rtxSupported = true
-                        logger.cdebug { "RTX payload type signaled, enabling RTX probing" }
-                    }
-                    if (pt is VideoPayloadType) {
-                        videoPayloadTypes.add(pt)
-                    }
-                }
+        object : MapEventValueFilterHandler<Byte, PayloadType>({ it is VideoPayloadType }) {
+            override fun entryAdded(key: Byte, value: PayloadType) {
+                videoPayloadTypes[key.toInt()] = (value as VideoPayloadType)
             }
-        }
+
+            override fun entryUpdated(key: Byte, newValue: PayloadType) {
+                videoPayloadTypes[key.toInt()] = (newValue as VideoPayloadType)
+            }
+
+            override fun entryRemoved(key: Byte) {
+                videoPayloadTypes.remove(key.toInt())
+            }
+        }.apply { streamInformationStore.onRtpPayloadTypeEvent(this.mapEventHandler) }
```

Comment:
i played with another way of doing these sorts of filters using a helper method, let me know if you find this more readable:

```
watchForWhenValue(
    { it is VideoPayloadType },
    entryAdded = { key: Byte, value: PayloadType -> videoPayloadTypes[key.toInt()] = (value as VideoPayloadType) },
    entryRemoved = { key: Byte -> videoPayloadTypes.remove(key.toInt()) },
    entryUpdated = { key: Byte, value: PayloadType -> videoPayloadTypes[key.toInt()] = (value as VideoPayloadType) }
).apply { streamInformationStore.onRtpPayloadTypeEvent(this.mapEventHandler) }
```
i like that the  `watchForWhenValue` makes the predicate more  obvious,  but i don't like the style of passing the methods in (as arguments instead of implementing the class) as much.

Reply:
I find the style that's in the PR easier to read. Although I would remove the usage of `apply` in favor of a helper `val`. I like `apply`, but I find I have to spend some efford to understand what it's doing (what is `this.mapEventHandler`? Ah, yes, the `this` reference got changed.), and the overhead is not worth it when the block is a single line.

Also, in my mental model `MapEventValueFilterHandler` gets translated to "this thing with the very long name that starts with Map".

Reply:
Actually, where is the definition of `MapEventValueFilterHandler`? The compiler doesn't find it.

Reply:
yes, sorry, i forgot to commit that file.  added now

Reply:
> Also, in my mental model MapEventValueFilterHandler gets translated to "this thing with the very long name that starts with Map".

I think the `Handler` part could get dropped.  I wanted to include 'event' so as to convey it filters events, not the map itself (which to me suggests something different).  `MapEventValueFilter`?

Reply:
> I find the style that's in the PR easier to read. Although I would remove the usage of apply in favor of a helper val. I like apply, but I find I have to spend some efford to understand what it's doing (what is this.mapEventHandler? Ah, yes, the this reference got changed.), and the overhead is not worth it when the block is a single line.

k, will add a helper val.

Reply:
> I think the `Handler` part could get dropped. I wanted to include 'event' so as to convey it filters events, not the map itself (which to me suggests something different). `MapEventValueFilter`?

Yes, I think that's better. After seeing how the class works it all makes sense (and it's nifty).